src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -397,250 +337,352 @@ export default function parseManifest(
+                adaptation.attributes.maxPlayoutRate;
+            }
+
+            if (representation.attributes.maximumSAPPeriod != null) {
+              parsedRepresentation.maximumSAPPeriod =
+                representation.attributes.maximumSAPPeriod;
+            } else if (adaptation.attributes.maximumSAPPeriod != null) {
+              parsedRepresentation.maximumSAPPeriod =
+                adaptation.attributes.maximumSAPPeriod;
+            }
+
+            if (representation.attributes.mimeType != null) {
+              parsedRepresentation.mimeType =
+                representation.attributes.mimeType;
+            } else if (adaptation.attributes.mimeType != null) {
+              parsedRepresentation.mimeType =
+                adaptation.attributes.mimeType;
+            }
+
+            if (representation.attributes.profiles != null) {
+              parsedRepresentation.profiles =
+                representation.attributes.profiles;
+            } else if (adaptation.attributes.profiles != null) {
+              parsedRepresentation.profiles =
+                adaptation.attributes.profiles;
+            }
+
+            if (representation.attributes.qualityRanking != null) {
+              parsedRepresentation.qualityRanking =
+                representation.attributes.qualityRanking;
+            }
+
+            if (representation.attributes.segmentProfiles != null) {
+              parsedRepresentation.segmentProfiles =
+                representation.attributes.segmentProfiles;
+            } else if (adaptation.attributes.segmentProfiles != null) {
+              parsedRepresentation.segmentProfiles =
+                adaptation.attributes.segmentProfiles;
+            }
+
+            if (representation.attributes.width != null) {
+              parsedRepresentation.width =
+                representation.attributes.width;
+            } else if (adaptation.attributes.width != null) {
+              parsedRepresentation.width =
+                adaptation.attributes.width;
+            }
+
+            return parsedRepresentation;
+          });
 
-          // 4-2-5. Add optional attributes
-          let codecs : string|undefined;
-          if (representation.attributes.codecs != null) {
-            codecs = representation.attributes.codecs;
-          } else if (adaptation.attributes.codecs != null) {
-            codecs = adaptation.attributes.codecs;
+        const adaptationMimeType = adaptation.attributes.mimeType;
+        const adaptationCodecs = adaptation.attributes.codecs;
+
+        const representationMimeTypes = representations
+          .map(r => r.mimeType)
+          .filter((mimeType : string|undefined) : mimeType is string => mimeType != null);
+
+        const representationCodecs = representations
+          .map(r => r.codecs)
+          .filter((codecs : string|undefined) : codecs is string => codecs != null);
+
+        const type: AdaptationType = inferAdaptationType(
```

Comment:
Why the double casting here?

Either inferAdaptationType should be typed to return only AdaptationType.

Either period.adaptations, should be a ``Dictionnary<Adaptation[]>`` and type filtering done elsewhere (``manifest/factory``? ``manifest/period``?)